syntax = "proto3";

package vector_service.v1;

import "shared_proto/v1/error.proto";
import "google/protobuf/struct.proto";

service VectorSearchService {
  rpc Search(SearchRequest) returns (SearchResponse);
}

message SearchRequest {
  optional string query_text = 1;
  bytes image = 2;
  optional string entity_type = 3;
  optional int32 limit = 4;
  google.protobuf.Struct filters = 5;
}

message SearchData {
  string id = 1;
  double similarity_score = 2;
  string payload_json = 3;
}

message SearchResponse {
  repeated SearchData data = 1;
  shared_proto.v1.ErrorDetails error = 2;
}
